lua/lazyvim/plugins/extras/lang/rust.lua

```diff
@@ -37,7 +47,13 @@ return {
   {
     "williamboman/mason.nvim",
     optional = true,
-    opts = { ensure_installed = { "codelldb" } },
+    opts = function(_, opts)
+      opts.ensure_installed = opts.ensure_installed or {}
+      vim.list_extend(opts.ensure_installed, { "codelldb" })
+      if diagnostics == "bacon-ls" then
+        vim.list_extend(opts.ensure_installed, { "bacon", "bacon-ls" })
```

Comment:
You could also remove `bacon-ls` from `mason.ensure_installed`, since LSP servers that are defined in `nvim-lspconfig` spec get installed automatically if they are `enabled`. This is just a suggestion and doesn't mean you absolutely have to do it. It's just how the rest of LazyVim LSP servers are defined as well throughout the codebase. [Corresponding code snippet in LazyVim](https://github.com/LazyVim/LazyVim/blob/a33eabddd9f3786421652d07894516d946c88647/lua/lazyvim/plugins/lsp/init.lua#L218-L231).

Reply:
`bacon` needs to be remain if I am not mistaken, since it's not an lsp server itself..

Reply:
Yes correct.